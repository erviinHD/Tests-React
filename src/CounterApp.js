import React, { useState } from "react";
import PropTypes from 'prop-types';

const CounterApp = ({ number = 1 }) => {

    const [counter, setCounter] = useState(number);

    //handeAdd
    const handleAdd = (e) => {
        setCounter(counter + 1);
    }

    const handleSubtract = (e) => {
        setCounter(counter - 1);
    }

    const handleReset = (e) => {
        setCounter(number);
    }

    return (
        <>
            <h1>CounterApp</h1>
            <p>{counter}</p>

            <button onClick={handleAdd}> +1 </button>
            <button onClick={handleReset}> Reset </button>
            <button onClick={handleSubtract}> -1 </button>
        </>)
}


CounterApp.propTypes
    = {
    number: PropTypes.number.isRequired
}

export default CounterApp;
