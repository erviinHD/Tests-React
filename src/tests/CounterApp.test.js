import React from 'react';

import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { shallow, configure } from 'enzyme';
import CounterApp from '../CounterApp';

configure({ adapter: new Adapter() });


describe('Pruebas en CunterApp', () => {
    const numer = 5;
    let input = shallow(<CounterApp number={numer} />);

    beforeEach(() => {
        input = shallow(<CounterApp number={numer} />);
    })

    test('debe mostrar el componente <CounterApp />', () => {

        expect(input).toMatchSnapshot();
    })

    test('Mostrar el contador', () => {
        const x = 200

        const wrapper = shallow(<CounterApp number={x} />);

        const textParrafo = wrapper.find('p').text();

        expect(Number(textParrafo)).toBe(x)
    })


    test('Debe incrementar el contador', () => {
        input.find('button').at(0).simulate('click');
        const counterText = input.find('p').text().trim();

        expect(counterText).toBe('6')

    })

    test('Debe decrementar el contador', () => {
        input.find('button').at(2).simulate('click');
        const counterText = input.find('p').text().trim();
        expect(counterText).toBe('4')

    })

    test('Debe colocar el valor por defecto de reset', () => {
        const wrapper = shallow(<CounterApp number={105} />);

        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(1).simulate('click');

        const counterText = wrapper.find('p').text().trim();

        expect(counterText).toBe('105')

    })
})