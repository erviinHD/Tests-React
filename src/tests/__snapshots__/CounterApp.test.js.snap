// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en CunterApp debe mostrar el componente <CounterApp /> 1`] = `ShallowWrapper {}`;
